### DISCUSSION TITLE
getting:  Tycho build extension not configured for MavenProject (when upping from 2.6 to 2.75)

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-tycho/tycho#1525] [General]
We want to move to Java17 with our product so 2.6 didn't work for me (i got stuff like compiler level not supported)
so i moved to Type 2.7.5 (i tried first 3.0.0 but that had other problems like i think already is discussed here, i will try this later and report it)
For many of our git repo's this all worked fine it compiles nicely.

But then i hit our main, which is the actual eclipse based project/git, that really is git repo that has the feature/product/target stuff.

When i just want to build only the target platform plugin i already get this: 

[ERROR] Failed to execute goal org.eclipse.tycho:target-platform-configuration:2.7.5:target-platform (default-target-platform) on project com.servoy.eclipse.target: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:2.7.5:target-platform failed: Tycho build extension not configured for MavenProject: com.servoy:com.servoy.eclipse.target:2022.12.0.3820_rc @ 

I do have that .mvn/extensions.xml with the contents:

```
<extensions>
 <extension>
        <groupId>org.eclipse.tycho</groupId>
        <artifactId>tycho-build</artifactId>
        <version>${tycho.version}</version>
    </extension>
</extensions>
```

i also tried it without..

The error doesn't really say where to look or what todo.

**Comment by jcompagner** [comment]
hmm it does build in that git repo my "nomal" java plugins
(if i do them one by hand)

the one that fails is my target one:

```
<project xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>com.servoy</groupId>
		<artifactId>servoy-eclipse</artifactId>
		<version>2022.12.0.3820_rc</version>
	</parent>
  <artifactId>com.servoy.eclipse.target</artifactId>
  <packaging>eclipse-target-definition</packaging>

</project>
```

so all other have the same parent (and those build fine)
but that i guess <packaging>eclipse-target-definition</packaging> doesn't work for me

**Comment by laeubi** [comment]
> Tycho build extension not configured for MavenProject: 

Usually means you need to add Tycho as a build plugin with `<extension>true</extension>`:
https://github.com/eclipse-tycho/tycho/blob/ac83caac1d94c795d5c62ee9d84f538549048bb9/demo/itp01/pom.xml#L28-L33

**Comment by jcompagner** [comment]
yes i already have that for years.. 

the weird thing is it is something about the parent pom...
Because as i said i couldn't build my target one (but i  could build other normal java plugins or feature projects)
But the one that generates a target failed..
Now i remove the parent pom reference in the target, so my target is just:

```
<project xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.servoy</groupId>
  <version>2022.12.0.3820_rc</version>
  <artifactId>com.servoy.eclipse.target</artifactId>
  <packaging>eclipse-target-definition</packaging>
    <properties>
        <tycho.version>2.7.5</tycho.version>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <build>
        <plugins>
            <plugin>
                <groupId>org.eclipse.tycho</groupId>
                <artifactId>tycho-maven-plugin</artifactId>
                <version>${tycho.version}</version>
                <extensions>true</extensions>
            </plugin>
        </plugins>
    </build>
</project>
```

then it builds but if i then go to the parent i try to build:

[INFO] Building servoy-eclipse 2022.12.0.3820_rc                         [2/25]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ servoy-eclipse ---
[INFO]
[INFO] --- target-platform-configuration:2.7.5:target-platform (default-target-platform) @ servoy-eclipse ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] com.servoy.eclipse.target 2022.12.0.3820_rc ........ SUCCESS [  0.247 s]
[INFO] servoy-eclipse 2022.12.0.3820_rc ................... FAILURE [  0.106 s]

that servoy-eclipse is the parent pom..

and that fails:

[ERROR] Failed to execute goal org.eclipse.tycho:target-platform-configuration:2.7.5:target-platform (default-target-platform) on project servoy-eclipse: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:2.7.5:target-platform failed: Tycho build extension not configured for MavenProject: com.servoy:servoy-eclipse:2022.12.0.3820_rc

with the target-platform-configuration:

but if i then go to a sub folder and build that (that has the parent pom) i get:

INFO] --- target-platform-configuration:2.7.5:target-platform (default-target-platform) @ com.servoy.eclipse.model ---
[INFO] Maven Artifact com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava is not a bundle and will be ignored, automatic wrapping of such artifacts can be enabled with <pomDependencies>wrapAsBundle</pomDependencies> in target platform configuration.
[INFO] Maven Artifact com.google.errorprone:error_prone_annotations:2.11.0 is not a bundle and will be ignored, automatic wrapping of such artifacts can be enabled with <pomDependencies>wrapAsBundle</pomDependencies> in target platform configuration.
[INFO] Maven Artifact com.google.j2objc:j2objc-annotations:1.3 is not a bundle and will be ignored, automatic wrapping of such artifacts can be enabled with <pomDependencies>wrapAsBundle</pomDependencies> in target platform configuration.

not sure why i get those google (guava) warnings here, but then it works nicely and builds it..

so in other works the target platform which was very simple pom but a reference to the parent pom failed.
Making that standalone without a ref to the parent pom then it worked.
But building only a feature project or a java plugin that also references the parent pom works fine...

**Comment by jcompagner** [comment]
that building of sub folders or that parent works by the way in the mode that the subfolder doesn't have a pom.xml at all or it does have 1 pointing to the parent.

**Comment by jcompagner** [comment]
it is just quite weird because this parent pom doesn't want to build:

https://github.com/Servoy/servoy-eclipse/blob/master/pom.xml

but it is very similar build up as:

https://github.com/Servoy/servoy-client/blob/master/pom.xml

and that one works fine..


(these are just the "old" one that use tycho 2.6.0, i have some changes now locally)

**Reply by laeubi** [reply]
This does not looks correct:

https://github.com/Servoy/servoy-eclipse/blob/97a9303d89be700840545237fbf8b2d704e494d4/pom.xml#L188-L195

what do you want to archive with that?

**Reply by jcompagner** [reply]
ha, thx.. that seems to be it...

That i think is there to sign the exe (so the eclipse.exe )  file that is generated by our own certificate 
so that starts here:

https://github.com/Servoy/servoy-eclipse/blob/master/com.servoy.eclipse.product/pom.xml#L57

(and the lines below that like that jsign mvn plugin)

but i need to have a look why i needed that in the parent pom.. (tycho 2.6 didn't seem to bother)
